### ISSUE TITLE
Split list of speakers functionality from agenda item

### ISSUE THREAD

**Issue by normanjaeckel** [OpenSlides/OpenSlides#4550]
Many users request a more flexible handling of the agenda (e.g. remove all motions from agenda without loosing list of speakers feautre).

So I propose to switch back to the following structure: Motiosn, Assignments, MotionBlocks and maybe other objetcs get the list of speakers feautre by respective fields (using a model mixin). The agenda item is reduced to explicit agenda features like number, tree, time, comment (but *not* title and text). If a new motion (or assignment or block) is created, it is not automatically added to the agenda but this can be configured (add new motions yes or not, add new topics always yes, ...)

This would be a bigger change, maybe for v3.1

**Comment by normanjaeckel** [comment]
@emanuelschuetze Please let me know if we want to try this. @ others: What do you think?

**Comment by ostcar** [comment]
The list of speakers could be a root rest element.

I would like to talk about this. We changed the structure so many times. Maybe we remember why we changed it to the system it is today.

**Comment by FinnStutzenstein** [comment]
@ostcar I think, if we make everything configurable, one could achive the current behavior. But if a user wants to e.g. just have list of speakers and no agenda, this is also possible. Do you see some draw backs anywhere?

Optional agenda items are very nice. It keeps the agenda clean, so it does not get crowded with items, that are not wanted. It is a better solution then the current three-state-visibility which causes headaches when sorting a **filtered** agenda, in fact no one wants to sort an agenda with over 300 entries.

So on creation, one can choose to add an agenda item for the content object. I thought of this config field with the following options (global for every possible content object):

 - always add an agenda item [option not visible on creation]
 - never add an agenda item [option not visible on creation]
 - let the user decide, default true
- let the user decide, default false

Also a user can add and remove an agenda item afterwards (like a button 'put in agenda'/'remove from agenda'). When deleting an agenda item the user could be asked, if we wants to remove from agenda, or also delete the content object. This should not be too confusing, I guess.

The list of speakers must be available without an agenda item, so it cannot be a "child rest element" of an item. My first idea is to just divide the agenda item model: `speaker_list_closed` and `speakers` will move to a seperate model which also gets a `content_object`. This object gets always created (like the current agenda items). Another, more close approach, would be to have a "speaker-Mixin", that adds the first meantioned fields to every model, e.g. motions and topics.

(As I meantioned topics: there an agenda item should be necessary)

Any thoughts on this?

**Comment by FinnStutzenstein** [comment]
We discussed this from the client's perspective and from managing permissions with the meantioned two options: A speaker mixin or an own speaker model.

The conclusion was to better use an own model, like the agenda item is an own model and not a mixin (which would be possible but many mixed fields and a complicated restriction process). If you want more information about the choosen way, please ask. For the same reason we want to split the `Item` model and make a `Speaker` model. They share some common fields like `content_object` and `title_information` but can be controlled seperately. In the first step, both models will be always created (as agendeitems now). Both creations may be limited later on.

I'll do some first work for this.

**Comment by ostcar** [comment]
I don't get it. If the agenda item and the speaker is separated, can there be speakers without an agenda item? What is the usecase? When does a speaker speak, that is not on the agenda?

Or do you want do add one speaker to many agenda items? I thought a speaker was the relation between a user and a agenda item. When a speaker can speak to many agenda items, what is the difference between a user and a speaker?

The only meaningful relation between a speaker and an agenda item I can see is a one-to-one relation. The only reason I see for a one-to-one relation is, that the two models are to big, that an autoupdate should be separated.

In conclusion, I don't understand at all, what you are trying to accomplish.

**Comment by FinnStutzenstein** [comment]
We do have usecases where a speakerlist exist, but no agenda item for it. In a second case the delegates should not see the agenda, but can have access to the list of speakers. By seperating items and speakers the usage is more flexible in the way of not tying the agenda and list of speakers together.
@emanuelschuetze Will add more information about this.

**Comment by normanjaeckel** [comment]
For me it is interesting why you prefer an extra root rest model for the speaker (or the list of speakers?) instead of a mixin. But we can discuss this also on phone call. For me it is not an urgent issue.

**Comment by FinnStutzenstein** [comment]
Already done
